### PR TITLE
feat(app-framework): Add UseSession attribute to replace annotation

### PR DESCRIPTION
**Pull request by ChristophWurst**
* Resolves: the urge to do something fancy

## Summary

https://help.nextcloud.com/t/how-should-we-use-php8-attributes/104278

## TODO

- [x] Add new attribute
- [x] Adjust session middleware
- [x] Migrate core controllers

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
  - [x] https://github.com/nextcloud/documentation/pull/9589
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: tests/lib/AppFramework/Middleware/SessionMiddlewareTest.php [hunk 1]
```diff
@@ -14,84 +17,130 @@
...
 use OCP\AppFramework\Http\Response;
+use OCP\IRequest;
+use OCP\ISession;
+use PHPUnit\Framework\MockObject\MockObject;
+use Test\TestCase;
 
-class SessionMiddlewareTest extends \Test\TestCase {
-	/** @var ControllerMethodReflector */
+class SessionMiddlewareTest extends TestCase {
+	/** @var ControllerMethodReflector|MockObject */
```
Comment (zak39) [inline tests/lib/AppFramework/Middleware/SessionMiddlewareTest.php:28]: Hi @ChristophWurst :)

Do you think the union types could work in this case ? 🤔 

https://www.php.net/releases/8.0/en.php#union-types

Comment (ChristophWurst) [reply tests/lib/AppFramework/Middleware/SessionMiddlewareTest.php:28]: They work indeed. Thanks for the tip!

Comment (zak39) [reply tests/lib/AppFramework/Middleware/SessionMiddlewareTest.php]: You're welcome !! ;)

**Review by rullzer — APPROVED** [review]
Fancy pancy

**Review by miaulalala — APPROVED** [review]
clueless

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
The autoloaders are not up to date
Please run: bash build/autoloaderchecker.sh

**Comment by nickvergessen** [comment]
Please document in https://github.com/nextcloud/server/issues/34692

**Pull request merged**